can't compile with GCC 6

tested with gcc 6.3.1, 6.4.0

`g++-6 -o test.o -c -std=c++17 test.cpp`

```
boost/sml.hpp:1029:38: error: ‘auto’ parameter not permitted in this context
   template <class TMappings, auto... Ns, class sm_impl, class State, class TEvent, class TDeps, class TSubs, class... TStates>
                                      ^~
boost/sml.hpp: In static member function ‘static bool boost::sml::v1_1_0::back::policies::fold_expr::dispatch_impl(sm_impl&, State&, boost::sml::v1_1_0::aux::index_sequence<Ns ...>, const TEvent&, TDeps&, TSubs&, const boost::sml::v1_1_0::aux::type_list<T7s ...>&)’:
boost/sml.hpp:1032:31: error: ‘Ns’ was not declared in this scope
     return ((current_state == Ns
                               ^~
```

--- a/include/boost/sml.hpp
+++ b/include/boost/sml.hpp
@@ -1026,7 +1026,7 @@ struct switch_stm {
 };
 #if defined(__cpp_fold_expressions)
 struct fold_expr {
-  template <class TMappings, auto... Ns, class sm_impl, class State, class TEvent, class TDeps, class TSubs, class... TStates>
+  template <class TMappings, int... Ns, class sm_impl, class State, class TEvent, class TDeps, class TSubs, class... TStates>
   static bool dispatch_impl(sm_impl &self, State &current_state, aux::index_sequence<Ns...>, const TEvent &event, TDeps &deps,
                             TSubs &subs, const aux::type_list<TStates...> &) {
     return ((current_state == Ns

--- a/include/boost/sml/back/policies/dispatch.hpp
+++ b/include/boost/sml/back/policies/dispatch.hpp
@@ -76,7 +76,7 @@ struct switch_stm {
 
 #if defined(__cpp_fold_expressions)  // __pph__
 struct fold_expr {
-  template <class TMappings, auto... Ns, class sm_impl, class State, class TEvent, class TDeps, class TSubs, class... TStates>
+  template <class TMappings, int... Ns, class sm_impl, class State, class TEvent, class TDeps, class TSubs, class... TStates>
   static bool dispatch_impl(sm_impl &self, State &current_state, aux::index_sequence<Ns...>, const TEvent &event, TDeps &deps,
                             TSubs &subs, const aux::type_list<TStates...> &) {
     return ((current_state == Ns

